fiftyone/core/utils.py

```diff
@@ -3143,3 +3143,26 @@ def validate_hex_color(value):
 
 
 fos = lazy_import("fiftyone.core.storage")
+
+
+@contextmanager
+def async_executor(
```

Comment:
All public methods in core modules should get full docstrings (see other methods in module). Can you add that please? 😄